app/views/session.py

```diff
@@ -53,13 +57,18 @@ def login():
         cookie_session.clear()
 
     decrypted_token = decrypt_token(request.args.get('token'))
+
     validate_jti(decrypted_token)
 
-    claims = parse_runner_claims(decrypted_token)
+    try:
+        claims = validate_runner_claims(decrypted_token)
+    except ValidationError as e:
+        logger.exception('Invalid claims on login')
+        raise InvalidTokenException('Invalid claims') from e
 
     g.schema = load_schema_from_metadata(claims)
     schema_metadata = g.schema.json['metadata']
-    validate_metadata(claims, schema_metadata)
+    claims = validate_questionnaire_claims(claims, schema_metadata)
```

Comment:
What happens if there are missing claims? Previously this raised a `InvalidTokenException`